pr-check.js

```diff
@@ -0,0 +1,260 @@
+const async = require("async")
+,   notification = require('./notification')
+,   w3ciprcheck = require('./w3c-ipr')
+,   doAsync = require('doasync') // rather than utils.promisy to get "free" support for object methods
+,   w3c = require("node-w3capi");
+
+let store, log;
+
+async function findW3CUserFromGithub(user) {
+  log.info("Looking for github user with id " + user.ghID + " in W3C API");
+  try {
+    // can't easily promisify w3c API :(
+    let w3cuser = await new Promise((res, rej) => w3c.user({type: 'github', id: user.ghID}).fetch((err, u) => { if (err) return rej(err); return res(u);}));
+    log.info(JSON.stringify(w3cuser, null, 2));
+    await doAsync(store).mergeOnUser(user.username, {
+      w3cid:  w3cuser.id,
+      w3capi: w3cuser._links.self.href.replace(/.*\//, "")
+    });
+  } catch (err) {
+    return user;
+  }
+  log.info("Found matching W3C user");
+  return doAsync(store).getUser(user.username);
+}
+
+async function findOrCreateUserFromGithub(username, gh) {
+  let user;
+  try {
+    user = await doAsync(store).getUser(username);
+  } catch (err) {
+    if (err.error !== "not_found") throw err;
+  }
+  if (!user) {
+    log.info("Getting GH id from github for " + username);
+    let ghuser = await gh.getUser(username);
+    // we store this for sake of efficiency
+    await doAsync(store).addUser(ghuser);
+    return findW3CUserFromGithub(ghuser);
+  } else {
+    // Let's check if the link has since been established
+    if (!user.w3capi) {
+      return findW3CUserFromGithub(user);
+    } else {
+      return user;
+    }
+  }
+}
+
+
+async function getStoredPR(fullname) {
+  log.info("Setting status for PR " + fullname);
+  let repo = await doAsync(store).getRepo(fullname);
+  if (!repo) throw ("Unknown repository: " + fullname);
+  let token = await doAsync(store).getToken(repo.owner);
+  if (!token) throw ("Token not found for: " + repo.owner);
+  return {repoGroups: repo.groups, token};
+}
+
+async function updateStoredPR(pr) {
+  log.info("Setting status for PR " + pr.fullName);
+  let updatedPr = await doAsync(store).updatePR(pr.fullName, pr.num, pr);
+  return updatedPr;
+}
+
+async function setGhStatus(gh, status) {
+  try {
+    await doAsync(gh).status(status);
+  } catch(err) {
+    log.error(err);
+  }
+}
+
+async function checkPrScope(gh, pr) {
+  const ignoreFiles = ["package.json", "package-lock.json", ".travis.yml", "w3c.json", "CONTRIBUTING.md", "LICENSE.md", "CODE_OF_CONDUCT.md"];
```

Comment:
```suggestion
  const ignoreFiles = ["package.json", "package-lock.json", ".travis.yml", "w3c.json", "CONTRIBUTING.md", "LICENSE.md", "LICENSE.txt", "CODE_OF_CONDUCT.md"];
```
from https://help.github.com/en/github/creating-cloning-and-archiving-repositories/licensing-a-repository#determining-the-location-of-your-license